flink-runtime/src/test/java/org/apache/flink/runtime/io/network/api/writer/RecordWriterTest.java

```diff
@@ -51,43 +51,31 @@
 import org.apache.flink.types.IntValue;
 import org.apache.flink.util.XORShiftRandom;
 
-import org.junit.Assert;
-import org.junit.Rule;
-import org.junit.Test;
-import org.junit.rules.TemporaryFolder;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
 
+import java.io.File;
 import java.io.IOException;
 import java.util.ArrayDeque;
 import java.util.Random;
 
 import static org.apache.flink.runtime.io.network.partition.PartitionTestUtils.createPartition;
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertFalse;
-import static org.junit.Assert.assertNotNull;
-import static org.junit.Assert.assertTrue;
+import static org.assertj.core.api.Assertions.assertThat;
 
 /** Tests for the {@link RecordWriter}. */
-public class RecordWriterTest {
+class RecordWriterTest {
 
-    private final boolean isBroadcastWriter;
+    boolean isBroadcastWriter = false;
```

Comment:
It's always false, right?

Reply:
@1996fanrui, if we keep that two constructor in master code, it will cause the exception: `Test class can only have one constructor error while running JUnit 5.`
Is there any suggestion to modify?

Reply:
How about adding a `boolean isBroadcastWriter()` mothod and removing all constructors and the `isBroadcastWriter` filed?


// Add the `boolean isBroadcastWriter()` to `RecordWriterTest`, it returns `false`.

```
    protected boolean isBroadcastWriter() {
        return false;
    }
```

// Add the `isBroadcastWriter()` to `BroadcastRecordWriterTest`, it returns `true`.

```
    protected boolean isBroadcastWriter() {
        return true;
    }
```

All callers call the method instead of filed. I have tried it on my Mac, it works well.

![](https://github.com/apache/flink/assets/38427477/a6f0a786-0745-4252-bca0-6e3f2a7cb830)

Reply:
Thanks @1996fanrui, that's a good idea.